### DISCUSSION TITLE
Question re --bytecode compilation

### DISCUSSION THREAD

**Discussion by rgillan** [oven-sh/bun#14422] [Q&A] [answered]
Have used the --bytecode option in 1.1.30 to compile one of our single file executables.  One of our hopes with this was to not have our source code contained within the single file executable (for distribution).  We have noticed that without the bytecode compile, one of our apps is 96.7MB, but when compiled to bytecode it goes up to 108.3MB.  Not so concerned about the size, but if we run
```strings appName```
we still see the various source files appended at the end.  Is there any way we could suppress/not have these in the resultant binary single file executable?

**Comment by Jarred-Sumner** [comment, marked as answer]
yeah the bytecode compilation in JSC does need the source code to work. It's not useful for obscuring the source.

**Reply by Jarred-Sumner** [reply]
you could use --minify to obscure the sourcecode somewhat

**Reply by rgillan** [reply]
Thanks @Jarred-Sumner that certainly makes it a little more difficult to read and not as obvious.  Might have to think about other approaches pre bun build to see what we can do, cheers and thanks for the responsiveness as always /r

**Reply by tiagoefreitas** [reply]
@rgillan what solution did you end up using?

**Reply by rgillan** [reply]
@tiagoefreitas we haven't got a solution for this as yet.  Luckily we are not shipping code publicly or via a public repo so the binary as is works (for now).  The --minify option does hide it somewhat.
